source/slang/slang-check-expr.cpp

```diff
@@ -2371,19 +2386,105 @@ namespace Slang
         return result;
     }
 
+    Expr* SemanticsExprVisitor::convertToLogicOperatorExpr(InvokeExpr* expr, bool& isArgumentsChecked)
+    {
+        LogicOperatorShortCircuitExpr* newExpr = nullptr;
+        isArgumentsChecked = false;
+
+        // If the logic expression is inside the generic parameter list, it cannot support short-circuit
+        // which will generate the ifelse branch.
+        if (!m_shouldShortCircuitLogicExpr)
+        {
+            return nullptr;
+        }
+
+        const ReflectClassInfo& classInfo = expr->functionExpr->getClassInfo();
+        const ASTNodeType astType = ASTNodeType(classInfo.m_classId);
+        if (ASTNodeType::VarExpr == astType)
+        {
+            VarExpr const* varExpr = as<VarExpr const>(expr->functionExpr);
+            if ((varExpr->name->text == "&&") || (varExpr->name->text == "||"))
+            {
+                // Don't use short-circuiting in differentiable function
+                if (getParentDifferentiableAttribute())
```

Comment:
Why can't we use short-circuiting in differentiable function?

Reply:
Done removing this.